src/main/webapp/app/entities/file-upload-exercise/file-upload-exercise.route.ts

```diff
@@ -26,54 +63,45 @@ export const fileUploadExerciseRoute: Routes = [
         canActivate: [UserRouteAccessService],
     },
     {
-        path: 'course/:courseId/file-upload-exercise',
-        component: FileUploadExerciseComponent,
-        data: {
-            authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
-            pageTitle: 'artemisApp.textExercise.home.title',
+        path: 'course/:courseId/file-upload-exercise/new',
+        component: FileUploadExerciseUpdateComponent,
+        resolve: {
+            fileUploadExercise: FileUploadExerciseResolve,
         },
-        canActivate: [UserRouteAccessService],
-    },
-    {
-        path: 'course/:courseId/file-upload-exercise/:id',
-        component: FileUploadExerciseDetailComponent,
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
-            pageTitle: 'artemisApp.textExercise.home.title',
+            pageTitle: 'artemisApp.fileUploadExercise.home.title',
         },
         canActivate: [UserRouteAccessService],
     },
-];
-
-export const fileUploadExercisePopupRoute: Routes = [
     {
-        path: 'course/:courseId/file-upload-exercise-new',
-        component: FileUploadExercisePopupComponent,
+        path: 'file-upload-exercise/:exerciseId/edit',
+        component: FileUploadExerciseUpdateComponent,
+        resolve: {
+            fileUploadExercise: FileUploadExerciseResolve,
+        },
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
             pageTitle: 'artemisApp.fileUploadExercise.home.title',
         },
         canActivate: [UserRouteAccessService],
-        outlet: 'popup',
     },
     {
-        path: 'file-upload-exercise/:id/edit',
-        component: FileUploadExercisePopupComponent,
+        path: 'course/:courseId/file-upload-exercise',
+        component: FileUploadExerciseComponent,
         data: {
             authorities: ['ROLE_TA', 'ROLE_INSTRUCTOR', 'ROLE_ADMIN'],
             pageTitle: 'artemisApp.fileUploadExercise.home.title',
         },
         canActivate: [UserRouteAccessService],
-        outlet: 'popup',
     },
     {
-        path: 'file-upload-exercise/:id/delete',
-        component: FileUploadExerciseDeletePopupComponent,
+        path: 'course/:courseId/file-upload-exercise/:id',
```

Comment:
Shouldn't it be `course/:courseId/file-upload-exercise/:exerciseId` to be consistent?

Reply:
Yes, good point